github-runner-provisioner/internal/codemagic/runner.go

```diff
@@ -0,0 +1,75 @@
+package codemagic
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	log "github.com/sirupsen/logrus"
+	"io"
+	"net/http"
+)
+
+type codemagicWorkflow struct {
+	WorkflowID  string                       `json:"workflowId"`
+	AppId       string                       `json:"appId"`
+	Branch      string                       `json:"branch"`
+	Environment map[string]map[string]string `json:"environment"`
+}
+
+func CreateMacM1Runner(ctx context.Context, owner string, repo string, token string, labels string, dryRun bool, codeMagicToken string) error {
+	client := &http.Client{}
+
+	data := codemagicWorkflow{
+		WorkflowID: "github-runner",
+		AppId:      "649493225428a76bc935a44b",
```

Comment:
Hardcoded ?

Reply:
Yep - this is the ID of the workflow in CodeMagic. We could pull it into a config layer if we wanted to, but I'm not sure if that's better or just adds unneeded complexity for this.